src/Microsoft.ML.FastTree/Dataset/DenseIntArray.cs

```diff
@@ -70,13 +70,14 @@ public override IntArray[] Split(int[][] assignment)
         }
 
 #if USE_FASTTREENATIVE
-        [DllImport("FastTreeNative", CallingConvention = CallingConvention.StdCall)]
+        internal const string NativePath = "FastTreeNative";
+        [DllImport(NativePath)]
         private static extern unsafe int C_Sumup_float(
```

Comment:
Do you need `CallingConvention = CallingConvention.Cdecl` here?

Reply:
No, everything except MKL should work via default calls.
MKL should use cdecl.

---
In reply to: [239656885](https://github.com/dotnet/machinelearning/pull/1833#discussion_r239656885) [](ancestors = 239656885)